Tasks/Common/VstsAzureHelpers_/InitializeFunctions.ps1

```diff
@@ -643,24 +665,22 @@ function Disconnect-AzureAndClearContext {
         if ($authScheme -eq 'ServicePrincipal') {
             Write-Verbose "Trying to disconnect from Azure and clear context"
 
-            if (Get-Command -Name "Disconnect-AzureRmAccount" -ErrorAction "SilentlyContinue") {
-                Write-Host "##[command]Disconnect-AzureRmAccount"
-                $null = Disconnect-AzureRmAccount
+            if (Get-Command -Name "Disconnect-AzureRmAccount" -ErrorAction "SilentlyContinue" -and CmdletHasMember -cmdlet Disconnect-AzureRmAccount -memberName Scope) {
+                Write-Host "##[command]Disconnect-AzureRmAccount -Scope Process -ErrorAction Stop"
+                $null = Disconnect-AzureRmAccount -Scope Process -ErrorAction Stop
             }
-            elseif (Get-Command -Name "Remove-AzureRmAccount" -ErrorAction "SilentlyContinue") {
-                Write-Host "##[command]Remove-AzureRmAccount"
-                $null = Remove-AzureRmAccount
+            elseif (Get-Command -Name "Remove-AzureRmAccount" -ErrorAction "SilentlyContinue" -and CmdletHasMember -cmdlet Remove-AzureRmAccount -memberName Scope) {
+                Write-Host "##[command]Remove-AzureRmAccount -Scope Process -ErrorAction Stop"
+                $null = Remove-AzureRmAccount -Scope Process -ErrorAction Stop
             }
-            elseif (Get-Command -Name "Logout-AzureRmAccount" -ErrorAction "SilentlyContinue") {
-                Write-Host "##[command]Logout-AzureRmAccount"
-                $null = Logout-AzureRmAccount
+            elseif (Get-Command -Name "Logout-AzureRmAccount" -ErrorAction "SilentlyContinue" -and CmdletHasMember -cmdlet Logout-AzureRmAccount -memberName Scope) {
+                Write-Host "##[command]Logout-AzureRmAccount -Scope Process -ErrorAction Stop"
+                $null = Logout-AzureRmAccount -Scope Process -ErrorAction Stop
             }
 
             if (Get-Command -Name "Clear-AzureRmContext" -ErrorAction "SilentlyContinue") {
-                Write-Host "##[command]Clear-AzureRmContext -Scope Process"
-                $null = Clear-AzureRmContext -Scope Process
-                Write-Host "##[command]Clear-AzureRmContext -Scope CurrentUser -Force -ErrorAction SilentlyContinue"
-                $null = Clear-AzureRmContext -Scope CurrentUser -Force -ErrorAction SilentlyContinue
+                Write-Host "##[command]Clear-AzureRmContext -Scope Process -Scope Process -ErrorAction Stop"
+                $null = Clear-AzureRmContext -Scope Process -ErrorAction Stop
```

Comment:
Any specific reason to move from SilentlyContinue to Stop

Reply:
just to catch the error message in the debug logs.